### PR TITLE
Add Code of Conduct, PR & Issue Templates to Improve Workflow

### PR DESCRIPTION
**Pull request by Byrdman32**
### **Description**
This Pull Request adds a Code of Conduct, Pull Request template, and multiple Issue templates to the Autonomy Software repository. These changes aim to establish clear community guidelines, streamline contribution processes, and improve issue tracking consistency.

### **Changes Made**
- Added `CODE_OF_CONDUCT.md`, outlining community standards and expectations for behavior based on the Contributor Covenant, version 2.1.
- Added a default Pull Request template to guide contributors when opening a new PR.
- Added the following Issue templates:
  - Bug Report
  - Enhancement Request
  - Research Interest
  - Package Upgrade

### **How These Changes Work**
1. **Code of Conduct**: Applies to all project interactions, providing a structure for reporting and handling violations, and maintaining a respectful and welcoming community.
2. **Pull Request Template**: Provides a standardized format for contributors to follow when submitting PRs, ensuring that all relevant information is included.
3. **Issue Templates**: Ensures that issues are properly categorized and that the necessary information is provided when reporting bugs, or requesting features.

### **Testing**
- Reviewed the content of the `CODE_OF_CONDUCT.md` to ensure clarity and alignment with the Contributor Covenant.
- Verified that the PR and issue templates are correctly formatted and appear in the GitHub UI when creating new issues or PRs.

### CODE REVIEW

File: INSTALL.md [hunk 1]
```diff
@@ -1,52 +1,95 @@
...
-- Use the searchbar to find the `Dev Containers` extension and click the install button.
-- Feel free to install any other useful extensions that might help you develop your code.
+- Go to the extensions tab (the little square icon on the left toolbar) and search for `Dev Containers`.
+- Click the install button to add it to your toolkit.
 
 ![](data/README_Resources/images/vscode_install_extensions.png)
 
-NOTE: In the later steps when we start the devcontainer, It's extensions will be separate from the extensions locally installed on your computer. You can still install extensions to the devcontainer, but the extensions and their settings won't be persistent between container rebuild. If you would like an extension to come packaged in with the container, ask one of our [software leads](https://github.com/orgs/MissouriMRDT/teams/software_leads).
+**Note:** Once we’re in the devcontainer, the extensions you install locally will be separate from the ones inside the container. Want a particular extension to be included by default? Hit up one of our [software leads](https://github.com/orgs/MissouriMRDT/teams/software_leads).
```
Comment (ClayJay3) [inline INSTALL.md]: Software leads link is broken.

File: README.md [hunk 2]
```diff
@@ -40,102 +53,256 @@
...
+If tests are available, they will be compiled into separate executables for unit and integration tests. These tests can be run individually or using CTest.
 
-- **states**: Contains C++ files related to the state machine code of our Rover. These files define the different states and transitions that govern the Rover's behavior, allowing for efficient task execution and adaptability to changing conditions.
+---
 
-- **util**: Contains utility scripts and helper functions used in our Rover project. These files provide various tools and functionalities to support development, debugging, and project management tasks, increasing overall efficiency and code reusability.
+## 🗒️ Third-Party Libraries
 
-- **vision**: Contains C++ files related to vision processing or computer vision algorithms used in our Rover. These files enable the Rover to process visual data, such as images or videos, and extract meaningful information for autonomous navigation and object recognition.
+This project makes use of several open-source libraries and tools that help us achieve our autonomy goals. We would like to acknowledge and thank the developers of the following libraries:
```
Comment (ClayJay3) [inline README.md:240]: Consider adding Eigen for consistency.

File: README.md [hunk 3]
```diff
@@ -40,102 +53,256 @@
...
-    cmake -DBUILD_SIM_MODE=OFF ..
-    make -j8
-    ./Autonomy_Software
-    ```
+- **Doxygen** - Tool for generating documentation from annotated C++ sources.  
+  [Doxygen GitHub](https://github.com/doxygen/doxygen)  
+  License: [GPL-2.0 License](https://github.com/doxygen/doxygen/blob/master/LICENSE)
 
-2) **Simulation Mode:** Which operates using the Webots Robot Simulator and uses the RoveComm manifest but sends all packets to local host instead. This allows the robot simulator to retrieve all the packets and run the simulation. To build and run Autonomy in this mode use the following commands:
+---
```
Comment (ClayJay3) [inline README.md:297]: Maybe add one more subsection that lists and points to submodule repos like threadpool and rovecomm.

**Review by ClayJay3 — CHANGES_REQUESTED** [review]
Docs are looking awesome! Nice overhaul!

File: README.md (hunk 3 above)
Comment (ClayJay3) [reply README.md:297]: A quick explination of rovecomm might be good to throw in somewhere too.

File: README.md [hunk 4]
```diff
@@ -40,102 +53,256 @@
...
 
-- **tools**: Contains files that provide utility scripts, development tools, or miscellaneous functionalities for our project. These files assist in various development tasks, such as data analysis, visualization, or simulation, enhancing the overall development experience.
+- **PyTorch** - A deep learning framework for building and training neural networks.  
+  [PyTorch GitHub](https://github.com/pytorch/pytorch)  
+  License: [BSD-3-Clause License](https://github.com/pytorch/pytorch/blob/main/LICENSE)
 
-- **external**: Contains external dependencies, libraries, or third-party modules used in our Rover project. These files are essential for the project's functionality and are organized based on their purpose or source, ensuring proper integration and compliance with licensing requirements.
+- **TensorFlow** - Open-source library for machine learning and artificial intelligence.  
+  [TensorFlow GitHub](https://github.com/tensorflow/tensorflow)  
+  License: [Apache License 2.0](https://www.tensorflow.org/license)
```
Comment (ClayJay3) [inline README.md]: Change this to tensorflow lite and libedgetpu
**TensorFlow Lite**: [TensorFlow Lite Documentation](https://www.tensorflow.org/lite/guide)
**Libedgetpu**: [Edge TPU Documentation](https://coral.ai/docs/reference/cpp/)

**Comment by ClayJay3** [comment]
Also consider adding the new .md docs as a new link in the doxygen layout so that they appear on the doxygen site.

File: README.md (hunk 3 above)
Comment (Byrdman32) [reply README.md:297]: I'll at some point add an explaination to what RoveComm is as part of the submodule but for now there isn't a good place in the docs that I can see.

File: INSTALL.md (hunk 1 above)
Comment (Byrdman32) [reply INSTALL.md]: This has apparently been broken for a few months now since I renamed the team for consistancy

**Review by ClayJay3 — APPROVED** [review]

**Pull request merged**